src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (beast instanceof Carnivore || beast.hidden) {
+      return;
+    }
+
+    beast.health -= 50;
+
+    if (beast.health <= 0) {
```

Comment:
This condition is not nesesary. You also can filter without it.

Reply:
Well, maybe you are right. Slightly more filtering, but less code. Might be a fair tradeoff. Changed!